lms/services/h_api.py

```diff
@@ -163,6 +171,24 @@ def _api_request(self, method, path, body=None, headers=None, stream=False):
 
         return response
 
+    def get_userid(self, username):
+        """Return the h userid for the given username and authority."""
+        return HUser(username).userid(self._authority)
+
+    def get_username(self, userid):
+        """
+        Return the h username for the given userid.
+
+        For example if userid is 'acct:seanh@hypothes.is' then return 'seanh'.
+
+        :raises ValueError: if the given userid isn't valid
+        """
+        match = re.match(r"^acct:([^@]+)@(.*)$", userid)
```

Comment:
I was checking if the second part of the regex could be tighter but looks like we do the same thing on H:


https://github.com/hypothesis/h/blob/main/h/util/user.py